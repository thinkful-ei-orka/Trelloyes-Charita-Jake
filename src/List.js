import React from 'react';
import Card from './card.js';
import './List.css';


export default function List(props) {

    const listOfCards = props.cards.map(card => (
      <Card key={card.id} title={card.title} content={card.content} />
    ));
    console.log(Card);
    return (
        <section className="List">
            <header className="List-header">
             <h3>{props.header}</h3>
            </header>
            <div className="List-cards">
             {listOfCards}
            </div>
    </section>  

    );
  }


