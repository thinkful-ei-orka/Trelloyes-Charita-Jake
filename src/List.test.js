import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

const testHeader = 'This is a test header';
const testCards = 
[{ id: 'a', title: 'First card', content: 'lorem ipsum' },
{ id: 'b', title: 'Second card', content: 'lorem ipsum' }];

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header={testHeader} cards={testCards}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<List header={testHeader} cards={testCards} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

