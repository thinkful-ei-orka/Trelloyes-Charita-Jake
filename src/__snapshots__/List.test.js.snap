// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h3>
      This is a test header
    </h3>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
